common/set.go

```diff
@@ -0,0 +1,33 @@
+package common
+
+import "encoding/json"
+
+// Set enables deduplication of objects
+type Set[T any] struct {
```

Comment:
This approach is better than using `UniqueMap` that was added in https://github.com/OpsLevel/opslevel-jq-parser/pull/34. I explain why in that PR description.